derive_builder/CHANGELOG.md

```diff
@@ -6,6 +6,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Added
 - customize setter names via `#[builder(setter(name="..."))]`
+- customize build_fn name via `#[builder(build_fn(name="..."))]`
+- suppress build method generation via `#[builder(build_fn(skip))]`
+- perform pre-build validation via `#[builder(build_fn(validator="path::to::fn"))]`
```

Comment:
@faern What do you think about the attribute nesting?
* should it be **nested** like `build_fn(validate_fn)`, e.g. 
  `#[builder(build_fn(name="finish", validate_fn="Self::validate"))]`
* or should it be **flat** like `build_fn, validate_fn`, e.g.
  `#[builder(build_fn(name="finish"), validate_fn="Self::validate")]`

Usually we do nesting, to organize settings. But we already have some kind of exception with `default` which currently only affects the `build_fn` but is not nested. However `default` is also a field specific option and _may_ also affect getters in the future. The reason there was that `#[builder(build_fn(default))]` would have looked a bit weird on a field level. So the situation is definitely a bit different with `validate_fn`.

I think my current favourite would be `validate` + nested, e.g. `#[builder(build_fn(validate="Self::validate"))]`. This way many attributes could be read as imperative verbs, like in _please `validate` this build function via `Self::validate`_, _please `skip` this_, _please `default` this_, _please `name` this `foo`_, ...

It looks like we have a split, where one person each is in favour of `validator`, `validate_fn` or `validate`. ^^